### PR TITLE
docs: Restructure the "Architecture Design" Section

### LINKED ISSUE #6456: docs: Restructure the "Architecture Design" Section
### Improved or typo fixed.

The current "[Architecture Design](https://apisix.apache.org/docs/apisix/architecture-design/route)" pages misses internal hyperlinks and could require some rewriting to enhance readability.

### Solution

Make changes to the docs similar to this PR: https://github.com/apache/apisix-website/pull/893

Please assign this issue to me. I will open a PR to discuss the changes.

### PR DESCRIPTION
**Pull request by navendu-pottekkat**
Signed-off-by: Navendu Pottekkat <navendupottekkat@gmail.com>

### What this PR does / why we need it:

Fixes #6456 

This PR adds internal hyperlinks and rewrites the Architecture Design section of the docs.

This improves the readability of this section of the docs.

### CODE REVIEW

**Comment by navendu-pottekkat** [comment]
@Anita-ihuman @juzhiyuan @avinal Could you please review this PR? I will fix the lint errors.

File: docs/en/latest/architecture-design/apisix.md [hunk 1]
```diff
@@ -55,6 +59,8 @@ etcd:
   host: "http://foo:2379" # etcd address
 ```
 
-Other default configurations can be found in the `conf/config-default.yaml` file, which is bound to the APISIX source code. **Never** manually modify the `conf/config-default.yaml` file. If you need to customize any configuration, you should update the `config.yaml` file.
+Default configurations of APISIX can be found in the `conf/config-default.yaml` file.
+
+**Note**: This file is bound to the APISIX source code and should **not** be modified. The configuration should only be changed by the methods mentioned above.
```
Comment (juzhiyuan) [inline docs/en/latest/architecture-design/apisix.md:64]:
```suggestion
**Note**: This file is bound to the APISIX source code and should **NOT** be modified. The configuration should only be changed by the methods mentioned above.
```

File: docs/en/latest/architecture-design/apisix.md [hunk 2]
```diff
@@ -55,6 +59,8 @@ etcd:
...
 
-**Note** `APISIX` will generate `conf/nginx.conf` file automatically, so please _DO NOT EDIT_ `conf/nginx.conf` file too.
+**Note**: The `conf/nginx.conf` file is automatically generated by APISIX and should **not** be edited.
```
Comment (juzhiyuan) [inline docs/en/latest/architecture-design/apisix.md:66]:
```suggestion
**Note**: The `conf/nginx.conf` file is automatically generated by APISIX and should **NOTE** be edited.
```

File: docs/en/latest/architecture-design/consumer.md [hunk 3]
```diff
@@ -21,32 +21,34 @@ title: Consumer
...
 | -------- | -------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
-| username | Yes      | Consumer Name.                                                                                                                                                                                   |
-| plugins  | No       | The corresponding plugin configuration of the Consumer, which has the highest priority: Consumer > Route > Service. For specific plugin configurations, refer to the [Plugins](plugin.md) section. |
+| username | Yes      | Name of the consumer.                                                                                                                                                                                   |
+| plugins  | No       | Plugin configuration of the Consumer. has the highest priority: Consumer > Route > Service. For specific Plugin configurations, refer the [Plugins](./plugin.md) section. |
```
Comment (juzhiyuan) [inline docs/en/latest/architecture-design/consumer.md:37]:
```suggestion
| plugins  | No       | Plugin configuration of the Consumer. It has the highest priority: Consumer > Route > Service. For specific Plugin configurations, refer the [Plugins](./plugin.md) section. |
```

File: docs/en/latest/architecture-design/plugin.md [hunk 4]
```diff
@@ -60,23 +64,28 @@ The plugin configuration is submitted as part of Route or Service and placed und
 }
 ```
 
-Not all plugins have specific configuration items. For example, there is no specific configuration item under `prometheus`. In this case, an empty object identifier can be used.
+Not all plugins have specific configuration items (for example, [prometheus](/docs/apisix/plugins/prometheus/)). In such cases, an empty object identifier can be used.
```
[the remark this answers is not in the document]
Comment (yzeng25) [reply docs/en/latest/architecture-design/plugin.md:67]: For the word "Plugin", it'd be good to unified them all as `Plugin` or `plugin`. I don't get the point when capitalizing  some of them.

File: docs/en/latest/architecture-design/route.md [hunk 5]
```diff
@@ -21,19 +21,25 @@ title: Route
 #
 -->
 
-The route matches the client's request by defining rules, then loads and executes the corresponding plugin based on the matching result, and forwards the request to the specified Upstream.
+Routes matches the client's request based on defined rules, loads and executes the corresponding [plugins](./plugin.md), and forwards the request to the specified [Upstream](./upstream.md).
```
Comment (yzeng25) [inline docs/en/latest/architecture-design/route.md:24]:
```suggestion
Routes match the client's request based on defined rules, loads and executes the corresponding [plugins](./plugin.md), and forwards the request to the specified [Upstream](./upstream.md).
```

File: docs/en/latest/architecture-design/apisix.md (hunk 2 above)
Comment (avinal) [reply docs/en/latest/architecture-design/apisix.md:66]:
```suggestion
**Note**: The `conf/nginx.conf` file is automatically generated by APISIX and should **NOT** be edited.
```

File: docs/en/latest/architecture-design/consumer.md [hunk 6]
```diff
@@ -21,32 +21,34 @@ title: Consumer
...
-3. Get the Plugin or Upstream information bound to the Consumer: Complete the different configurations for different Consumers.
-
-To sum up, Consumer is a consumer of certain types of services and needs to be used in conjunction with the user authentication system.
+1. The first step is Authentication. This is achieved by Authentication Plugins like [key-auth](../plugins/key-auth.md) and [JWT](../plugins/jwt-auth.md).
+2. After authenticating, you can obtain the `id` of the Consumer. This `id` will be the unique identifier of a Consumer.
+3. The configurations like Plugins and Upstream bound to the Consumer are then executed.
 
-For example, different consumers request the same API, and the gateway service corresponds to different Plugin or Upstream configurations according to the current request user information.
+Consumers are useful when you have different consumers requesting the same API and you need to execute different Plugin and Upstream configurations based on the consumer. These needs to be used in conjunction with the user authentication system.
```
Comment (avinal) [inline docs/en/latest/architecture-design/consumer.md:47]:
```suggestion
Consumers are useful when you have different consumers requesting the same API and you need to execute different Plugin and Upstream configurations based on the consumer. These need to be used in conjunction with the user authentication system.
```

File: docs/en/latest/architecture-design/plugin.md [hunk 7]
```diff
@@ -21,17 +21,21 @@ title: Plugin
 #
 -->
 
-`Plugin` represents the plugin configuration that will be executed during the `HTTP` request/response lifecycle.
+This represents the configuration of the plugins that are executed during the HTTP request/response lifecycle.
 
-The `Plugin` configuration can be bound directly to `Route` or it can be bound to `Service` or `Consumer`. For the configuration of the same plugin, only one copy is valid, and the configuration selection priority is always `Consumer` > `Route` > `Service`.
+A `Plugin` configuration can be bout directly to a [`Route`](./route.md), a [`Service`](./service.md) or a [`Consumer`](./consumer.md).
```
Comment (avinal) [inline docs/en/latest/architecture-design/plugin.md:26]:
```suggestion
A `Plugin` configuration can be bound directly to a [`Route`](./route.md), a [`Service`](./service.md) or a [`Consumer`](./consumer.md).
```

File: docs/en/latest/architecture-design/plugin.md [hunk 8]
```diff
@@ -21,17 +21,21 @@ title: Plugin
...
 
-In `conf/config.yaml`, you can declare which plugins are supported by the local APISIX node. This is a whitelisting mechanism. Plugins that are not in this whitelist will be automatically ignored. This feature can be used to temporarily turn off or turn on specific plugins, which is very effective in dealing with unexpected situations. If you want to add new plugins based on existing plugins, you need to copy the data of `plugins` node which in `conf/config-default.yaml` to the `plugins` node of `conf/config.yaml`.
+Note: While configuring the same plugin, only one copy of the configuration is valid. The order of precedence is always `Consumer` > `Route` > `Service`.
```
Comment (avinal) [inline docs/en/latest/architecture-design/plugin.md:28]:
```suggestion
**Note**: While configuring the same plugin, only one copy of the configuration is valid. The order of precedence is always `Consumer` > `Route` > `Service`.
```

File: docs/en/latest/architecture-design/plugin.md [hunk 9]
```diff
@@ -43,7 +47,7 @@ local _M = {
 }
 ```
 
-The plugin configuration is submitted as part of Route or Service and placed under `plugins`. It internally uses the plugin name as the hash's key to hold configuration items for different plugins.
+A Plugin configuration is submitted as part of the Route or Service and is placed under `plugins`. It internally uses the Plugin name as the hash's key to hold configuration items for the different Plugins.
```
Comment (avinal) [inline docs/en/latest/architecture-design/plugin.md:50]:
```suggestion
A Plugin configuration is submitted as part of the Route or Service and is placed under `plugins`. It internally uses the Plugin name as the hash key to hold the configuration items for the different Plugins.
```

File: docs/en/latest/architecture-design/router.md [hunk 10]
```diff
@@ -21,19 +21,23 @@ title: Router
...
-    - `match priority`: first try absolute match, if you can't hit absolute match, try prefix match.
-    - `Any filter attribute`: Allows you to specify any Nginx built-in variable as a filter, such as URL request parameters, request headers, cookies, and so on.
-  - `radixtree_uri_with_parameter`: Like `radixtree_uri` but also support parameter match.
-  - `radixtree_host_uri`: Use `host + uri` as the primary index (based on the `radixtree` engine), matching both host and URL for the current request.
+  - `radixtree_uri`: (Default) Only use the `uri` as the primary index. To learn more about the support for full and deep prefix matching, check [How to use router-radixtree](../router-radixtree.md).
+    - `Absolute match`: Match completely with the given `uri` (`/foo/bar`, `/foo/glo`).
+    - `Prefix match`: Match with the given prefix. Use `*` to represent the given `uri` for prefix matching. For example, `/foo*` can match with `/foo/`, `/foo/a` and `/foo/b`.
+    - `match priority`: First try absolute match, if it didn't match, try prefix matching.
```
Comment (avinal) [inline docs/en/latest/architecture-design/router.md:37]:
```suggestion
    - `match priority`: First try an absolute match, if it didn't match, try prefix matching.
```

File: docs/en/latest/architecture-design/upstream.md [hunk 11]
```diff
@@ -21,19 +21,21 @@ title: Upstream
 #
 -->
 
-Upstream is a virtual host abstraction that performs load balancing on a given set of service nodes according to configuration rules. Upstream address information can be directly configured to `Route` (or `Service`). When Upstream has duplicates, you need to use "reference" to avoid duplication.
+Upstream is a virtual host abstraction that performs load balancing on a given set of service nodes according to the configured rules.
+
+Although Upstream can directly configured to the [Route](./route.md) or [Service](./route.md), using an Upstream object is recommended when there is duplication as shown below.
```
Comment (avinal) [inline docs/en/latest/architecture-design/upstream.md:26]:
```suggestion
Although Upstream can be directly configured to the [Route](./route.md) or [Service](./route.md), using an Upstream object is recommended when there is duplication as shown below.
```

**Comment by navendu-pottekkat** [comment]
@avinal @juzhiyuan @yzeng25 I have made the changes as suggested. Could you please check?

**Review by yzeng25 — APPROVED** [review]
LGTM

**Review by juzhiyuan — APPROVED** [review]
LGTM

**Review by avinal — APPROVED** [review]

**Pull request merged**